### PR TITLE
Optimize label index serialization

### PR DESCRIPTION
**Pull request by ccollie**
This pull request updates the persistence format for the postings index to group label entries by their shared name prefix, optimizing on-disk storage and improving efficiency. The serialization and deserialization logic is updated to support this grouped encoding, and new tests are added to verify correctness and edge cases.

**Persistence Format Improvements**
- Group label index entries by their shared `name=` prefix during serialization, emitting the prefix once per group instead of per entry, which reduces redundancy and storage size. This groups all label values with the same name together, leveraging the schema-bound nature of label names. [[1]](diffhunk://#diff-965f9e5cbc6ef7abd3c288d323e4e8a1550ed057148a5f364291285f97a839eaR136-R163) [[2]](diffhunk://#diff-965f9e5cbc6ef7abd3c288d323e4e8a1550ed057148a5f364291285f97a839eaR172-R205)
- Update deserialization to read the new group directory format, reconstructing the original keys by concatenating group prefixes and entry suffixes. This ensures efficient and correct restoration of the label index from disk.

**Documentation**
- Update module-level documentation to describe the new grouped encoding design, removing the reference to the old "Option A" and clarifying the new on-disk format.

**Testing**
- Add tests to ensure that label names are written only once per group in the payload, and that roundtrips preserve all key shapes, including edge cases like values containing `=` or keys with no `=` at all.

### CODE REVIEW

File: src/series/index/persistence.rs [hunk 1]
```diff
@@ -157,12 +169,40 @@ fn serialize_postings_section(buf: &mut Vec<u8>, db: i32, postings: &Postings) {
...
-    write_uvarint(buf, entry_count);
+    if let Some((open, count)) = open_group {
+        write_byte_slice(&mut groups, open);
+        write_uvarint(&mut groups, count);
+        group_count += 1;
+    }
+
+    write_uvarint(buf, group_count);
+    buf.extend_from_slice(&groups);
     buf.extend_from_slice(&entries);
```
[the remark this answers is not in the document]
Comment (ccollie) [reply src/series/index/persistence.rs:205]: The library is unreleased, so there are no in-the-wild implementations to consider

File: src/series/index/persistence.rs [hunk 2]
```diff
@@ -817,6 +874,76 @@ mod tests {
         assert!(loaded.all_postings.contains(1) && loaded.all_postings.contains(3));
     }
 
+    fn count_occurrences(haystack: &[u8], needle: &[u8]) -> usize {
+        haystack
+            .windows(needle.len())
+            .filter(|w| *w == needle)
+            .count()
+    }
```
[the remark this answers is not in the document]
Comment (ccollie) [reply src/series/index/persistence.rs:881]: This exists strictly in test code as a utility. It is not executed in persistence code.

File: src/series/index/persistence.rs [hunk 3]
```diff
@@ -133,22 +133,34 @@ fn serialize_postings_section(buf: &mut Vec<u8>, db: i32, postings: &Postings) {
...
-    // drains continuously); entries that become empty are dropped, so the entry count is
-    // written after the entries are serialized.
+    // Keys are written grouped by their shared `name=` prefix, which is emitted once per group
+    // rather than once per entry — the on-disk analogue of the prefix sharing the ART gives us
+    // in memory. Label *value* cardinality is what grows (hosts, instances); the set of label
+    // *names* is small and schema-bound, so a name with k values costs one copy of the name
+    // instead of k. Sorted iteration order is what makes this a single pass: a group's entries
+    // are necessarily contiguous, because every key in it shares the prefix up to and including
+    // the first `=`, and no other group's prefix can fall between them (a prefix never contains
+    // `=` except as its last byte, so no prefix is a prefix of another).
```
[the remark this answers is not in the document]
Comment (ccollie) [reply src/series/index/persistence.rs:143]: The statement is correct, because the key/value creation and index insertion methods guarantee this on construction
